packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -47,22 +49,69 @@ class ImageCache {
     _maximumSize = value;
     if (maximumSize == 0) {
       _cache.clear();
+      _sizeBytes = 0;
     } else {
-      while (_cache.length > maximumSize)
-        _cache.remove(_cache.keys.first);
+      _checkCacheSize();
     }
   }
 
+  /// The current number of cached entries.
+  int get currentSize => _cache.length;
```

Comment:
Maybe call this length, for consistency with Map?

Reply:
oh but then you lose the symmetry with maximumSize.
So yeah, I'd keep this as you have it.